activerecord/lib/active_record/connection_adapters.rb

```diff
@@ -21,6 +21,17 @@ def register(name, class_name, path = class_name.underscore)
         @adapters[name.to_s] = [class_name, path]
       end
 
+      # Registers name alias for a database adapter.
+      #
+      # == Example
+      #
+      #   ActiveRecord::ConnectionAdapters.alias("trilogy", as: "mysql")
+      #
+      def alias(original, as:)
+        resolve(original) # will raise if original is invalid or does not exist
+        register(as, *@adapters[original.to_s])
```

Comment:
```suggestion
        register(as, *@adapters.fetch(original.to_s))
```
So we fail if we try to alias something that doesn't exist yet.

Reply:
This is unnecessary as the above `resolve(original)` line will raise if it is not found.

Reply:
Ah right, I missed it.